### ISSUE TITLE
when installing an extension, it appears installation succeeds even if validation fails

### ISSUE THREAD

**Issue by boneskull** [appium/appium#17791]
In `installViaNpm()`, we show a spinner while the installation happens, then report "success" (shows a green checkmark) once it's completed without error and the `package.json` is parseable.

Thereafter, we perform additional validation against the content of `package.json` (the `appium` field, mostly).  This step can fail.

Maybe it'd be better to wrap the entire install/validate operation in a spinner (or try something different) so that a user does not see installation has succeeded when, in fact, it has failed due to validation.

While it may _technically_ be true that "installation" succeeded, it does not matter to the end user, who did not experience success.

**Comment by boneskull** [comment]
This might be a morass, which is why I'm making the issue instead of just going to fix it.

**Comment by jlipps** [comment]
yeah i think we should wrap the entire operation, or have two spinner steps, one for install and one for validate

**Comment by subhashvkotra** [comment]
I am in favour of two spinner steps, it gives clearer picture to anyone trying.

**Comment by mykola-mokhnach** [comment]
Closed as fixed
